tests/cpu_bugs/meltdown.pm

```diff
@@ -0,0 +1,86 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: CPU BUGS on Linux kernel check
+# Maintainer: James Wang <jnwang@suse.com>
+
+package meltdown;
+use strict;
+use warnings;
+
+use base "consoletest";
+use base "Mitigation";
+use bootloader_setup;
+use ipmi_backend_utils;
+use power_action_utils 'power_action';
+use testapi;
+use utils;
+
+my $mitigations_list =
+  {
+    name                   => "meltdown",
+    CPUID                  => hex '20000000',
+    IA32_ARCH_CAPABILITIES => 1,                #bit0 -- RDCL_NO
+    parameter              => 'pti',
+    cpuflags               => ['pti'],
+    sysfs_name             => "meltdown",
+    sysfs                  => {
+        on      => "Mitigation: PTI",
+        off     => "Vulnerable",
+        auto    => "Mitigation: PTI",
+        default => "Mitigation: PTI",
+    },
+    dmesg => {
+        on      => "Kernel/User page tables isolation: enabled",
+        off     => "",
+        auto    => "Kernel/User page tables isolation: enabled",
+        default => "Kernel/User page tables isolation: enabled",
+    },
+    cmdline => [
+        "on",
+        "off",
+        "auto",
+    ],
+    lscpu => {
+        on   => "pti",
+        off  => "",
+        auto => "pti",
+    },
+  };
+
+sub new {
+    my ($class, $args) = @_;
+    #Help constructor distinguishing is our own test object or openQA call
+    if ($args eq $mitigations_list) {
```

Comment:
Could you help explain this strange judgement more clear? And why this will happen, is there potential openqa issue here?

Reply:
@coolo Hi sir, This implement seems like a helpless way.
I need to override a method in the parent class, but this class will be initialized by openQA, If I call new() method here, it will report an error.
I have to use parameters to distinguish which is the call to openqa, which is my own call, in order to generate different objects.